cmd/gitops/root/cmd.go

```diff
@@ -86,6 +90,38 @@ func RootCmd() *cobra.Command {
 				fmt.Fprintf(os.Stderr, "Error: %v\n", err)
 				os.Exit(1)
 			}
+
+			_, err = config.GetConfig(false)
+			if err != nil {
+				fmt.Println("To improve our product, we would like to collect analytics data. You can read more about what data we collect here: https://docs.gitops.weave.works/docs/feedback-and-telemetry/")
+
+				prompt := promptui.Prompt{
+					Label:     "Would you like to turn on analytics to help us improve our product",
+					IsConfirm: true,
+					Default:   "Y",
+				}
+
+				enableAnalytics := false
+
+				// Answering "n" causes err to not be nil. Hitting enter without typing
+				// does not return the default.
+				_, err := prompt.Run()
+				if err == nil {
+					enableAnalytics = true
+				}
+
+				seed := time.Now().UnixNano()
+
+				gitopsConfig := &config.GitopsCLIConfig{
+					UserID:    config.GenerateUserID(10, seed),
+					Analytics: enableAnalytics,
+				}
+
+				if err = config.SaveConfig(gitopsConfig); err != nil {
+					fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+					os.Exit(1)
```

Comment:
@ozamosi and should we exit on a save config error here?

Reply:
@ozamosi just changed this code to `_ = config.SaveConfig(gitopsConfig)` to ignore errors on config save silently here.